packages/grafana-ui/src/components/uPlot/config/UPlotSeriesBuilder.ts

```diff
@@ -100,3 +118,47 @@ export class UPlotSeriesBuilder extends PlotConfigBuilder<SeriesProps, Series> {
     };
   }
 }
+
+function getCanvasGradient(
+  color: string,
+  gradientMode: AreaGradientMode,
+  opacity: number,
+  plotHeight: number
+): CanvasGradient {
+  const ctx = getCanvasContext();
+  const pxRatio = window.devicePixelRatio ?? 1;
+  const pixelHeight = (plotHeight - 30) * pxRatio;
```

Comment:
is 30 a rough fudge factor to exclude the axis & padding? feels like this should be done externally already, so getCanvasGradient doesn't need to care.

Reply:
Yes, but figured this is just temporary as the height is very wrong in other ways as well (does not account for legend below graph for example).

But if fill is changed to a function (callback) then should be able to get canvas height (excluding axes height) from the uplot instance I hope

Reply:
> But if fill is changed to a function (callback) then should be able to get canvas height (excluding axes height) from the uplot instance I hope

:100: